backend/db/refresh.py

```diff
@@ -35,22 +35,35 @@ def refresh_db(
     when the last refresh occurred, this will result in there being a gap in which any changes that occurred during that
     time will not be fetched, resulting in an incomplete database. Therefore by default this will raise an error unless
     this is set to True.
-    todo: Can update update_db_status_var() so that it can accept optional param 'con' to improve performance."""
+    todo: Can update update_db_status_var() so that it can accept optional param 'con' to improve performance.
+    todo: refactor `new_request_while_refreshing` usage for brevity in code and DB: Rather than checking a variable
+     new_request_while_refreshing , at the end of the refresh, if the last_refresh_request has changed / is newer than
+     what was set at the beginning of the script, it knows that a new request while refreshing has occurred, and use
+     that information to know to start new refresh. https://github.com/jhu-bids/TermHub/pull/469#discussion_r1253581138
+    """
     local = use_local_db
     print('INFO: Starting database refresh.', flush=True)  # flush: for gh action
     t0, t0_str = datetime.now(), current_datetime()
-    update_db_status_var('last_refresh_request', t0_str, local)
+
+    if check_db_status_var('refresh_status') == 'active':
+        update_db_status_var('new_request_while_refreshing', t0_str, local)
+        return
+
+    print('INFO: Starting database refresh.', flush=True)  # flush: for gh action
     update_db_status_var('refresh_status', 'active', local)
+    update_db_status_var('last_refresh_request', current_datetime(), local)
     try:
         with get_db_connection(local=local) as con:
             last_refresh = last_refresh_timestamp(con)
             if since and dp.parse(since) > dp.parse(last_refresh) and not force_non_contiguity:
                 raise ValueError(SINCE_ERR)
             since = since if since else last_refresh
 
-            # Refresh DB
-            # todo: will use instead when ready: all_new_objects_enclave_to_db()
-            csets_and_members_enclave_to_db(con, schema, since)
+            # Refresh DB:
+            # todo: when ready, will use all_new_objects_enclave_to_db() instead of csets_and_members_enclave_to_db()
+            if not csets_and_members_enclave_to_db(con, schema, since):
```

Comment:
### Closing out refresh if no new updates
I just noticed a few things that needed to be changed here!

1. Bugfix: `csets_and_members_enclave_to_db()` was actually always returning `None`!
The last line of the function was `return csets_and_members_to_db(con, schema, csets_and_members)`
But that function actually had no `return` statement. Basically when a function has no `return`, it's the same as that function returning `None`. So it was always returning `None`! I changed it to return `'success'` at the very end.

2. Updating some status variables
The way you had it, if `csets_and_members_enclave_to_db()` returned `None`, and then `refresh_db()` would exit out. But, this would result in a few of the DB variables not being updated. For example, `refresh_status` would still erroneously be in the state `active`. And I also felt it would be good to update `last_refresh_success` and `last_refresh_result`. So really, the only thing it doesn't do now when there were no new updates is that it doesn't call `counts_update()`.

3. Allowing it to do  `check_db_status_var('new_request_while_refreshing')`
So the way you had it, if nothing was returned `csets_and_members_enclave_to_db()`, it was exiting before it could get here. Realistically though, if there were no new database updates, and if the user clicked the button / there was another refresh request during that time, it is very likely that even if we check again, there will be no updates. Still though, it is possible that there could be some updates, and we also want to close out and at least run it and do `delete_db_status_var('new_request_while_refreshing')`,.